src/main/java/duke/Lister.java

```diff
@@ -0,0 +1,117 @@
+package duke;
+
+/**
+ * The Lister Class parses commands and references TaskList and Storage Classes to store and retrieve Tasks
+ * @author qiujingying
+ * @version 1.0
+ */
+public class Lister {
+    private TaskList taskList;
+    private Storage storage;
+
+    /**
+     * Creates a Lister object to reference to TaskList and Storage objects
+     * @param tasks initial TaskList stores Tasks
+     * @param storage initial storage with filepath to store data
+     */
+    public Lister(TaskList tasks, Storage storage) {
+        this.taskList = tasks;
+        this.storage = storage;
+    }
+
+    /**
+     * Parses commands from command line
+     * @param command input to identify relevant Task objects to create and their description
+     */
+    public void record(String command) {
+        if (command.equals("list")) {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < taskList.getSize(); i++) {
+                System.out.println((i + 1) + "." + taskList.retrieveTask(i).toString());
+            }
+        } else {
+            try {
+                int x = getIndex(command);
+                switch (command.substring(0, x)) {
+                case "delete":
+                    int b = Integer.valueOf(command.substring(x + 1)) - 1;
```

Comment:
What if the user keys in the wrong number (ie. a non-positive integer like zero, a float, or a negative number) / types of arguments (ie. string)? Or maybe the user keys in a positive integer but it does not fall within the indexes of the list. Perhaps you could look into handling such errors from the user.

Examples of wrong user inputs: delete 2 now !!!! (too many arguments and wrong types as well), delete now (wrong type of argument - got string instead of int), delete -2 (ArrayIndexOutOfBounds - AIOOB), delete 0 (AIOOB), delete 1000000000 (AIOOB), delete (not enough arguments)